[Bug]: Release process fail building OLM

### Is there an existing issue already for this bug?

- [x] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [x] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [x] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

_No response_

### Version

1.26.0-rc1

### What version of Kubernetes are you using?

1.32

### What is your Kubernetes environment?

Self-managed: kind (evaluation)

### How did you install the operator?

YAML manifest

### What happened?

The release process for rc2 failed building the OLM bundles due to the following error:
```
2025/04/16 20:40:32 Found pullspec for annotation postgresql-15.6-annotation: quay.io/enterprisedb/postgresql:15.6
170
2025/04/16 20:40:32 Found pullspec for annotation cloud-native-postgresql-latest-annotation: quay.io/enterprisedb/cloud-native-postgresql:latest
171
2025/04/16 20:40:32 Found pullspec for annotation cloud-native-postgresql-1.26.0-rc2-ubi9-annotation: quay.io/enterprisedb/cloud-native-postgresql:1.26.0-rc2-ubi9
172
2025/04/16 20:40:32 Found pullspec for annotation postgresql-16.2-annotation: quay.io/enterprisedb/postgresql:16.2
173
2025/04/16 20:40:32 Found pullspec for annotation postgresql-15.6-annotation: quay.io/enterprisedb/postgresql:15.6
174
2025/04/16 20:40:32 Found pullspec for annotation postgresql-16.2-annotation: quay.io/enterprisedb/postgresql:16.2
175
2025/04/16 20:40:32 Found pullspec for annotation postgresql-15.6-annotation: quay.io/enterprisedb/postgresql:15.6
176
time="2025-04-16T20:40:33Z" level=fatal msg="Error generating bundle manifests: error resolving image: parsing reference \"quay.io/enterprisedb/cloud-native-postgresql:1.26.0-rc2-ubi9,quay.io/enterprisedb/cloud-native-postgresql:latest\": could not parse reference: quay.io/enterprisedb/cloud-native-postgresql:1.26.0-rc2-ubi9,quay.io/enterprisedb/cloud-native-postgresql:latest"
177
make: *** [Makefile:154: olm-bundle] Error 1
178
Error: Process completed with exit code 2.
```
This means that we're not properly processing the json file with the image tag

### Cluster resource

```shell

```

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/.github/workflows/release-publish.yml
+++ b/.github/workflows/release-publish.yml
@@ -96,7 +96,7 @@ jobs:
       author_name: ${{ steps.build-meta.outputs.author_name }}
       author_email: ${{ steps.build-meta.outputs.author_email }}
       platforms: ${{ env.PLATFORMS }}
-      ubi_img: ${{ fromJSON(steps.bake-push.outputs.metadata)['ubi']['image.name'] }}
+      ubi_img: ${{ env.UBI_IMAGE }}
     steps:
       -
         name: Checkout
@@ -214,12 +214,22 @@ jobs:
         # See https://github.blog/security/supply-chain-security/safeguard-container-signing-capability-actions/
         # and https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml for more details on
         # how to use cosign.
-      - name: Sign images
+      -
+        name: Sign images
         run: |
           images=$(echo '${{ steps.bake-push.outputs.metadata }}' |
             jq -r '.[] | (."image.name" | sub(",.*";"" )) + "@" + ."containerimage.digest"'
           )
           cosign sign --yes ${images}
+      -
+        # Bake returns all the tags for a target's variant in a comma separated list.
+        # We only care about a single tag for OLM, so we remove the "latest" tag and
+        # pick the first entry in order from what's left in the list
+        name: Output images
+        env:
+          ubi_tags: ${{ fromJSON(steps.bake-push.outputs.metadata)['ubi']['image.name'] }}
+        run: |
+          echo "UBI_IMAGE=$(echo "$ubi_tags" | tr ',' '\n' | grep -v 'latest' | sed 's/^ *//g' | head -n 1)" >> $GITHUB_ENV
 
   olm-bundle:
     name: Create OLM bundle and catalog

--- a/.github/workflows/release-publish.yml
+++ b/.github/workflows/release-publish.yml
@@ -96,7 +96,7 @@ jobs:
       author_name: ${{ steps.build-meta.outputs.author_name }}
       author_email: ${{ steps.build-meta.outputs.author_email }}
       platforms: ${{ env.PLATFORMS }}
-      ubi_img: ${{ env.UBI_IMAGE }}
+      olm_img: ${{ steps.olm-image.outputs.olm_image }}
     steps:
       -
         name: Checkout
@@ -225,11 +225,12 @@ jobs:
         # Bake returns all the tags for a target's variant in a comma separated list.
         # We only care about a single tag for OLM, so we remove the "latest" tag and
         # pick the first entry in order from what's left in the list
-        name: Output images
+        name: Image for OLM
+        id: olm-image
         env:
           ubi_tags: ${{ fromJSON(steps.bake-push.outputs.metadata)['ubi']['image.name'] }}
         run: |
-          echo "UBI_IMAGE=$(echo "$ubi_tags" | tr ',' '\n' | grep -v 'latest' | sed 's/^ *//g' | head -n 1)" >> $GITHUB_ENV
+          echo "olm_image=$(echo "$ubi_tags" | tr ',' '\n' | grep -v 'latest' | sed 's/^ *//g' | head -n 1)" >> $GITHUB_OUTPUT
 
   olm-bundle:
     name: Create OLM bundle and catalog
@@ -271,11 +272,11 @@ jobs:
 
       - name: Set bundle variables
         env:
-          UBI_IMG: ${{ needs.release-binaries.outputs.ubi_img }}
+          OLM_IMG: ${{ needs.release-binaries.outputs.olm_img }}
         run: |
-          echo "CONTROLLER_IMG=${UBI_IMG}" >> $GITHUB_ENV
-          echo "BUNDLE_IMG=${UBI_IMG}-bundle" >> $GITHUB_ENV
-          echo "CATALOG_IMG=${UBI_IMG}-catalog" >> $GITHUB_ENV
+          echo "CONTROLLER_IMG=${OLM_IMG}" >> $GITHUB_ENV
+          echo "BUNDLE_IMG=${OLM_IMG}-bundle" >> $GITHUB_ENV
+          echo "CATALOG_IMG=${OLM_IMG}-catalog" >> $GITHUB_ENV
 
       - name: Create bundle
         env:

